app/controllers/work_packages/activities_tab_controller.rb

```diff
@@ -0,0 +1,247 @@
+  end
+
+  def create
+    call = create_notification_service_call
+
+    if call.success? && call.result
+      if call.result.initial?
+        # we need to update the whole item component for an initial journal entry
+        # and not just the show part as happens in the time based update
+        # as this part is not rendered for initial journal
+        update_item_component(call.result, state: :show, filter: params[:filter]&.to_sym || :all)
+      end
+      generate_time_based_update_streams(params[:last_update_timestamp], params[:filter])
+    end
+
+    respond_with_turbo_streams
+  end
+
+  def update
+    call = Journals::UpdateService.new(model: @journal, user: User.current).call(
+      notes: journal_params[:notes]
+    )
+
+    if call.success? && call.result
+      update_item_component(call.result, state: :show, filter: params[:filter]&.to_sym || :all)
+    end
+    # TODO: handle errors
+
+    respond_with_turbo_streams
+  end
+
+  def update_sorting
+    filter = params[:filter]&.to_sym || :all
+
+    call = Users::UpdateService.new(user: User.current, model: User.current).call(
+      pref: { comments_sorting: params[:sorting] }
+    )
+
+    if call.success?
+      # update the whole tab to reflect the new sorting in all components
+      # we need to call replace in order to properly re-init the index stimulus component
+      replace_via_turbo_stream(
+        component: WorkPackages::ActivitiesTab::IndexComponent.new(
+          work_package: @work_package,
+          filter:
+        )
+      )
+    end
+
+    respond_with_turbo_streams
+  end
+
+  private
+
+  def find_work_package
+    @work_package = WorkPackage.find(params[:work_package_id])
+  end
+
+  def find_project
+    @project = @work_package.project
+  end
+
+  def find_journal
+    @journal = Journal.find(params[:id])
+  end
+
+  def journal_sorting
+    User.current.preference&.comments_sorting || "desc"
+  end
+
+  def journal_params
+    params.require(:journal).permit(:notes)
+  end
+
+  def create_notification_service_call
+    ### taken from ActivitiesByWorkPackageAPI
+    AddWorkPackageNoteService
+      .new(user: User.current,
+           work_package: @work_package)
+      .call(journal_params[:notes],
+            send_notifications: !(params.has_key?(:notify) && params[:notify] == "false"))
+    ###
+  end
+
+  def update_item_component(journal, state: :show, filter: :all)
+    update_via_turbo_stream(
+      component: WorkPackages::ActivitiesTab::Journals::ItemComponent.new(
+        journal:,
+        state:,
+        filter:
+      )
+    )
+  end
+
+  def generate_time_based_update_streams(last_update_timestamp, filter)
+    filter = filter&.to_sym || :all
+    # TODO: prototypical implementation
+    journals = @work_package.journals
+
+    if filter == :only_comments
+      journals = journals.where.not(notes: "")
+    end
+
+    journals.where("updated_at > ?", last_update_timestamp).find_each do |journal|
```

Comment:
Wouldn't it be easier to just rerender all at this point?

Reply:
Then we would potentially loose an edit state